HW04.Rmd

```diff
@@ -28,34 +35,112 @@ library(tidyverse)    # load tidyverse packages, including ggplot2
 Knitr is part of the knitr package. Allows you to change column names and add captions and make pretty tables in your knitted document. Hint: set the argument format = "markdown"
 
 
-```{r month, warning=FALSE}
+```{r month, warning=FALSE, echo=F}
+
+#Create the data frame for gun deaths per month
+deaths_month <- guns %>%
+    group_by(month) %>%
+    count(month) %>%
+    rename(Deaths_per_month = n)
+
+#Turns out R has a predefined object called month.abb that contains the abbreviated month names, much like month.name. This makes very easy to change the numbers in the data frame for the actual month names. You only have to change the column type to a factor and voilà!
+deaths_month$month <- factor(month.abb[deaths_month$month], levels = month.abb, ordered = TRUE)
+
+
+kable(deaths_month, format = "markdown", col.names = c("Month", "Deaths per month"))
 
 ```
 
 
 ### Generating a bar chart with human-readable labels on the x-axis. That is, each month should be labeled "Jan", "Feb", "Mar" (full or abbreviated month names are fine), not `1`, `2`, `3`.
 
-```{r month_chart}
+```{r month_chart, echo=F}
+
+#Since deaths_month$month was already a factor with abbreviated month names (line 47), you only need to graph.
+#The instruction was for the total number of deaths per month, not per year, but the data frame contains observations from 2012 to 2014.
 
+deaths_month %>%
+    ggplot(aes(x = month, y = Deaths_per_month)) + 
+    geom_col(fill = "blue") + #Filled just to pop some color there, I don't like gray.
+        labs(title = "Number of gun deaths per month from 2012-2014", #The instruction was for the total number of deaths per month, not per year, but the data frame contains observations from 2012 to 2014.
+         x = "Month",
+         y = "Number of gun deaths",
+         caption = "Source: fivethirtyeight.com")
 ```
 
 ## Generating a bar chart that identifies the number of gun deaths associated with each type of intent cause of death. The bars should be sorted from highest to lowest values.
 
-```{r intent}
+```{r intent, echo=F}
+
+deaths_intent <- guns %>%
+    group_by(intent) %>%
+    count(intent, sort = TRUE) %>% #Sorted values from highest to lowest
+    rename(no_of_deaths = n)
+
+
+kable(deaths_intent, format = "markdown", col.names = c("Intent", "No. of deaths"))
+
+#Just to order the intent from highest to lowest in a factor
+deaths_intent$intent_ordered <- deaths_intent$intent
+deaths_intent$intent <- factor(deaths_intent$intent, levels = deaths_intent$intent_ordered, ordered = TRUE)
+
+
+ggplot(deaths_intent, aes(x = intent, y = no_of_deaths)) +
+    geom_col(fill = "red") + 
+    scale_y_log10() + #Accidental, Undetermined, and NA went almost unnoticed in a regular scale.

```

Comment:
I understand that the data points are vastly spread out, but in my opinion, when graphing information that would likely be disseminated to the general public and contains such simple info as "type of death" and "#", the log scale adds confusion. To an unknowing observer, it looks like these different situations are much more close in # than they truly are. Even though accidental, undetermined, and NA went unnoticed, I personally think scaling y continuously would better represent the distribution to a general audience.

Reply:
I agree, and I actually change this one a lot before pushing my repository because I was so indecisive. But know that I think of it, maybe adding the number on top of each bar without log scale may be more intuitive.